### ISSUE TITLE
`load': cannot load such file -- /usr/share/rubygems-integration/all/gems/bundler-1.17.3/exe/bundle (LoadError)

### ISSUE THREAD

**Issue by bcoles** [rapid7/metasploit-framework#11368]
`bundle install` with the latest version of Metasploit `master` from GitHub, on a fully updated Kali Rolling box, throws the following error:

```
Traceback (most recent call last):
	1: from /usr/local/bin/bundle:23:in `<main>'
/usr/local/bin/bundle:23:in `load': cannot load such file -- /usr/share/rubygems-integration/all/gems/bundler-1.17.3/exe/bundle (LoadError)
```

**To fix, run `gem update --system` prior to `bundle install`.**

**Comment by busterb** [comment]
Is there really anything we can do about this other than document it? Unfortunately the bundler/ruby ecosystem goes a little nuts on a semi-annual basis when it comes to major version upgrades.

**Comment by jmartin-r7** [comment]
Another option is `gem install bundler -v'~>1.13'` prior to `bundle install`.  

Hopefully rubygems and bundler will release versions with effective restrictions in place to provide users a clear path, but for now these two options are what we have.

**Comment by jmartin-r7** [comment]
Another complication seen on ruby `2.6.1` which ships with bundler `1.17.2` and Rubygems `3.0.1`.  If bundler `1.17.3` is installed startup will fail.

```
$ ./msfconsole
[*] Metasploit requires the Bundler gem to be installed
    $ gem install bundler
```

Solution is `gem uninstall bundler -v 1.17.3`.

**Comment by bcoles** [comment]
> 
> 
> Is there really anything we can do about this other than document it? Unfortunately the bundler/ruby ecosystem goes a little nuts on a semi-annual basis when it comes to major version upgrades.

I can offer a few terrible solutions:

1. Force update `rubygems` and/or `bundler`
2. Backport changes in `master` to Framework2
3. Bundle an entire ruby environment
4. I hear good things about ~NodeJS~ ~Python~ ~Rust~ Golang

Tagging:
* https://github.com/bundler/bundler/issues/6970
* https://github.com/bundler/bundler/issues/6227

**Comment by yo8sbf** [comment]
thanks for this error and i resolved so when i installed bundle  appeared in terminal "Don't run Bundler as root. Bundler can ask for sudo if it is needed, and
installing your bundle as root will break this application for all non-root
users on this machine.

**Comment by yo8sbf** [comment]
can i install it in root? or non-root that is my question

**Comment by bcoles** [comment]
> 
> 
> can i install it in root? or non-root that is my question

If you run `bundle` as `root` it may break the application for all non-root users. If you run `bundle` as a non-root user, it will prompt you to enter your password for `sudo`, when required, for installation of gems which require elevated privileges.

**Comment by busterb** [comment]
Have we gone enough months without a new major breaking rubygems/bundler release to call this closable?

**Comment by jmartin-r7** [comment]
Will start testing with latest bundler env on 2.5.5 and 2.6.3 to validate that breaking states are squashed at this point.  If all pass then we can start rolling out current versions and close this.

**Comment by ccondon-r7** [comment]
@jmartin-r7 Is this closable now?

**Comment by jmartin-r7** [comment]
Closed by #14197 update to bundler 2
